packages/@aws-cdk/integ-runner/lib/runner/runners.ts

```diff
@@ -423,28 +447,119 @@ export interface RunOptions {
    * @default false
    */
   readonly dryRun?: boolean;
+
+  /**
+   * If this is set to false then the stack update workflow will
+   * not be run
+   *
+   * The update workflow exists to check for cases where a change would cause
+   * a failure to an existing stack, but not for a newly created stack.
+   *
+   * @default true
+   */
+  readonly updateWorkflow?: boolean;
 }
 
 /**
  * An integration test runner that orchestrates executing
  * integration tests
  */
 export class IntegTestRunner extends IntegRunner {
-  constructor(options: IntegRunnerOptions) {
+  constructor(options: IntegRunnerOptions, destructiveChanges?: DestructiveChange[]) {
     super(options);
+    this._destructiveChanges = destructiveChanges;
+  }
+
+  /**
+   * When running integration tests with the update path workflow
+   * it is important that the snapshot that is deployed is the current snapshot
+   * from the upstream branch. In order to guarantee that, first checkout the latest
+   * (to the user) snapshot from upstream
+   *
+   * It is not straightforward to figure out what branch the current
+   * working branch was created from. This is a best effort attempt to do so.
+   * This assumes that there is an 'origin'. `git remote show origin` returns a list of
+   * all branches and we then search for one that starts with `HEAD branch: `
+   */
+  private checkoutSnapshot(): void {
```

Comment:
What happens when the snapshot hasn't been added to git yet?

Reply:
hmm good question. Currently it will throw a warning saying it couldn't checkout the snapshot directory. I think it might be best to just add some info to the README about iterating on _new_ tests, i.e. if you are working on a new test you should run with `--disable-update-workflow`.

If it's a new test (snapshot hasn't been added to git) you shouldn't be using the update workflow. And I don't think there is a good way for us to automatically figure this out since you could have commited the snapshot and then run the test.

Reply:
Added section to readme.